From: Stefano Rivera <stefanor@debian.org>
Date: Mon, 28 Apr 2025 07:32:26 -0400
Subject: No need to install wheel any more

setuptools has had a native bdist_wheel since 70.1.0:
https://setuptools.pypa.io/en/latest/history.html#v70-1-0

And Debian's python-virtualenv package removed support for
--bundle=wheel in 20.30.0+ds-2.

Forwarded: https://github.com/python-cffi/cffi/pull/165
---
 testing/cffi0/test_zintegration.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/testing/cffi0/test_zintegration.py b/testing/cffi0/test_zintegration.py
index 53f48eb..9830c51 100644
--- a/testing/cffi0/test_zintegration.py
+++ b/testing/cffi0/test_zintegration.py
@@ -16,12 +16,12 @@ def create_venv(name):
     try:
         # FUTURE: we should probably update this to use venv for at least more modern Pythons, and
         # install setuptools/pip/etc explicitly for the tests that require them (as venv has stopped including
-        # setuptools and wheel by default for newer versions).
+        # setuptools by default for newer versions).
         subprocess.check_call(['virtualenv', 
             #'--never-download', <= could be added, but causes failures
             # in random cases on random machines
                                '-p', os.path.abspath(sys.executable),
-                               '--wheel=bundle', '--setuptools=bundle',
+                               '--setuptools=bundle',
                                str(tmpdir)])
 
     except OSError as e:
@@ -86,7 +86,7 @@ def really_run_setup_and_program(dirname, venv_dir_and_paths, python_snippet):
         # there's a setuptools/easy_install bug that causes this to fail when the build/install occur together and
         # we're in the same directory with the build (it tries to look up dependencies for itself on PyPI);
         # subsequent runs will succeed because this test doesn't properly clean up the build- use pip for now.
-        subprocess.check_call((vp, '-m', 'pip', 'install', '.'), env=env)
+        subprocess.check_call((vp, '-m', 'pip', 'install', '.', '--no-use-pep517'), env=env)
         subprocess.check_call((vp, str(python_f)), env=env)
     finally:
         os.chdir(olddir)
